internal/fs/fs.go

```diff
@@ -100,6 +102,53 @@ func HasFilepathPrefix(path, prefix string) bool {
 	return true
 }
 
+// EqualPaths compares the paths passed to check if the are equivalent.
+// It respects the case-sensitivity of the underlaying filesysyems.
+func EqualPaths(p1, p2 string) (bool, error) {
```

Comment:
`EquivalentPaths` would probably be a better name here than `EqualPaths`, as it emphasizes that we are testing for logically equivalent paths rather than byte-identical ones (which is the whole point!)